tests/roles/cinder_adoption/tasks/main.yaml

```diff
@@ -0,0 +1,50 @@
+- name: deploy podified Cinder scheduler and API
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc patch openstackcontrolplane openstack --type=merge --patch '
+    spec:
+      cinder:
+        enabled: true
+        template:
+          databaseInstance: openstack
+          secret: osp-secret
+          cinderAPI:
+            replicas: 1
+            containerImage: quay.io/tripleozedcentos9/openstack-cinder-api:current-tripleo
+          cinderScheduler:
+            replicas: 1
+            containerImage: quay.io/tripleozedcentos9/openstack-cinder-scheduler:current-tripleo
+          cinderBackup:
+            replicas: 0 # backend needs to be configured
+            containerImage: quay.io/tripleozedcentos9/openstack-cinder-backup:current-tripleo
+          cinderVolumes:
+            volume1:
+              containerImage: quay.io/tripleozedcentos9/openstack-cinder-volume:current-tripleo
+              replicas: 0 # backend needs to be configured
+    '
+
+- name: wait for Cinder to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod --selector=component=cinder-scheduler -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+    oc get pod --selector=component=cinder-api -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
```

Comment:
as sanity check the syntax here is fine. the test fails at this point
https://logserver.rdoproject.org/23/46923/24/check/data-plane-adoption-github-rdo-centos-8-crc-centos-9-standalone/11fd2bf/logs/subnode-1/home/zuul-worker/test_minimal_out_2023-05-04T16:01:35UTC.log.txt.gz

```
fatal: [localhost]: FAILED! => {"attempts": 60, "changed": true, "cmd": "set -euxo pipefail\n\n\noc get pod --selector=component=cinder-scheduler -o jsonpath='{.items[0].status.phase}{\"\\n\"}' | grep Running\noc get pod --selector=component=cinder-api -o jsonpath='{.items[0].status.phase}{\"\\n\"}' | grep Running\n", "delta": "0:00:00.488966", "end": "2023-05-04 16:14:38.753254", "msg": "non-zero return code", "rc": 1, "start": "2023-05-04 16:14:38.264288", "stderr": "+ grep Running\n+ oc get pod --selector=component=cinder-scheduler -o 'jsonpath={.items[0].status.phase}{\"\\n\"}'\nerror: error executing jsonpath \"{.items[0].status.phase}{\\\"\\\\n\\\"}\": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:\n\ttemplate was:\n\t\t{.items[0].status.phase}{\"\\n\"}\n\tobject given to jsonpath engine was:\n\t\tmap[string]interface {}{\"apiVersion\":\"v1\", \"items\":[]interface {}{}, \"kind\":\"List\", \"metadata\":map[string]interface {}{\"resourceVersion\":\"\"}}", "stderr_lines": ["+ grep Running", "+ oc get pod --selector=component=cinder-scheduler -o 'jsonpath={.items[0].status.phase}{\"\\n\"}'", "error: error executing jsonpath \"{.items[0].status.phase}{\\\"\\\\n\\\"}\": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:", "\ttemplate was:", "\t\t{.items[0].status.phase}{\"\\n\"}", "\tobject given to jsonpath engine was:", "\t\tmap[string]interface {}{\"apiVersion\":\"v1\", \"items\":[]interface {}{}, \"kind\":\"List\", \"metadata\":map[string]interface {}{\"resourceVersion\":\"\"}}"], "stdout": "", "stdout_lines": []}
```

however when i logged into the held nodes from that job just now:

```
[zuul-worker@np0003799736 ~]$ oc get pod --selector=component=cinder-scheduler -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
Running
[zuul-worker@np0003799736 ~]$ oc get pod --selector=component=cinder-api -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
Running
```

The error in logs happens when there is nothing returned for the get pod --selector i.e.

```
[zuul-worker@np0003799736 ~]$ oc get pod --selector=component=no-such-thing-yet -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
error: error executing jsonpath "{.items[0].status.phase}{\"\\n\"}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].status.phase}{"\n"}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":""}}

```


so you'll need to guard this a bit or find some other way to express it.

I don't know why this happens for cinder specifically perhaps it takes longer to start. I see you have the same code in others https://github.com/openstack-k8s-operators/data-plane-adoption/blob/a919ba4e7430ea5ce0604ffc9bd1e0b78cc45a21/tests/roles/glance_adoption/tasks/main.yaml#L20 

```
- name: wait for Glance to start up
```

Reply:
The check worked for me locally and we do have the same for other services, so hopefully it's ok. The pod didn't get started at all (it's not listed in all_pods.txt).